archive_upload/handlers/dsmc_handlers.py

```diff
@@ -0,0 +1,803 @@
+        self.write_object(response_data)
+
+
+class GenChecksumsHandler(BaseDsmcHandler):
+
+    """
+    Handler for generating checksums for an archive before uploading to PDC.
+    """
+
+    def post(self, runfolder_archive):
+        """
+        Calculates the MD5 checksums for each file in the runfolder archive, before uploading to PDC.
+        Job is run in the background to be polled by the status endpoint.
+
+        :param runfolder_archive: Name of the runfolder archive
+        :returns: HTTP 202 if checksum job has started successfully, with a `job_id` to be used in later polling,
+                  HTTP 500 if an unexpected error was encountered
+        """
+        path_to_archive_root = os.path.abspath(self.config["path_to_archive_root"])
+        log_dir = os.path.abspath(self.config["log_directory"])
+        checksum_log = os.path.abspath(os.path.join(log_dir, "checksum.log"))
+
+        if not self._validate_runfolder_exists(runfolder_archive, path_to_archive_root):
+            msg = "Error when validating runfolder. {} is not found under {}".format(
+                runfolder_archive, path_to_archive_root)
+            raise ArchiveException(reason=msg, status_code=400)
+
+        path_to_archive = os.path.join(path_to_archive_root, runfolder_archive)
+        filename = "checksums_prior_to_pdc.md5"
+
+        cmd = "cd {} && /usr/bin/find -L . -type f ! -path './{}' -exec /usr/bin/md5sum {{}} + > {}".format(
+            path_to_archive, filename, filename)
+        log.info("Generating checksums for {}".format(path_to_archive))
+        log.debug("Will now execute command {}".format(cmd))
+        job_id = self.runner_service.start(
+            cmd, nbr_of_cores=1, run_dir=log_dir, stdout=checksum_log, stderr=checksum_log)
+
+        status_end_point = "{0}://{1}{2}".format(
+            self.request.protocol,
+            self.request.host,
+            self.reverse_url("status", job_id))
+
+        response_data = {
+            "job_id": job_id,
+            "service_version": version,
+            "link": status_end_point,
+            "state": State.STARTED}
+
+        self.set_status(202, reason="started processing")
+        self.write_object(response_data)
+
+
+class CreateDirHandler(BaseDsmcHandler):
+
+    """
+    Handler for creating an archive to upload.
+    """
+
+    @staticmethod
+    def _verify_unaligned(srcdir):
+        """
+        Check that the archive contains the `Unaligned` symlink when running on biotanks.
+        The link should point to a proper directory.
+
+        :param srcdir: The path to the archive which we should investigate
+        :return: True if `srcdir` contains a symlink `Unaligned` that points to a directory,
+                 False otherwise
+        """
+        log.debug("Validating Unaligned...")
+
+        unaligned_link = os.path.join(srcdir, "Unaligned")
+        unaligned_dir = os.path.abspath(unaligned_link)
```

Comment:
Not exactly sure what your question is about, but I'm guessing you're wondering what the difference between `unaligned_link` and `unaligned_dir` is? I.e. aren't they pointing to the same location? Yes, in most cases they will, but not when testing. In deployed mode `srcdir` will usually be an absolute path, but in testing it is a relative path vs snpeq-archive-upload's root dir. Sometimes I use `abspath()` just to align things so it will be the same across the border.

Reply:
Ah, yes exactly. I gathered that `abspath` would only make a difference if the path was relative and was wondering where relative paths come into play.